metricflow-semantics/metricflow_semantics/model/semantics/semantic_model_lookup.py

```diff
@@ -241,17 +242,23 @@ def _add_semantic_model(self, semantic_model: SemanticModel) -> None:
             semantic_models_for_dimension = self._dimension_index.get(dim.reference, []) + [semantic_model]
             self._dimension_index[dim.reference] = semantic_models_for_dimension
 
+            assert StructuredLinkableSpecName.from_name(dim.name).is_element_name, (
```

Comment:
Should we log this error instead of raising it? Worried it could be a breaking change, and this is triggered on init so it will get hit even if the incorrect object is never accessed.

Reply:
Good call, I'll make this change here and upstack for things which are in use.

Reply:
SG! We might want to make sure we merge the whole stack at once in that case, otherwise this error could end up deploying to MFS unexpectedly.